R/aggregate_by_matrix.R

```diff
@@ -0,0 +1,37 @@
+#' @rdname aggregate
+#'
+#' @export
+colMeansMat <- function(x, MAT)
+  colSumsMat(x, MAT)/colSums(MAT)
+
+#' @rdname aggregate
+#'
+#' @export
+colSumsMat <- function(x, MAT)
+  t(MAT) %*% x
+
+
+##' @param MAT An adjacency matrix that defines what features with
+##'     `nrow(MAT) == nrow(x)`
+##'
+##' @return [aggregate_by_matrix()] returns a `matrix` (or `Matrix`)
+##'     of dimensions `ncol(MAT)` and `ncol(x), with `dimnames` equal
+##'     to `colnames(x)` and `rownames(MAT)`.
+##'
+##' @rdname aggregate
+##'
+##' @export
+aggregate_by_matrix <- function(x, MAT, FUN, ...) {
+    if (!inherits(x, "matrix"))
+        stop("'x' must be a matrix.")
+    if (!inherits(MAT, "Matrix") & !inherits(MAT, "matrix"))
```

Comment:
You should use `&&` to skip this `if` statement if the first test is `TRUE` already, e.g. MAT is a `Matrix`.
```suggestion
    if (!inherits(MAT, "Matrix") && !inherits(MAT, "matrix"))
```
BTW: It would be good to set the most common class in front here. Do you think most user will use `Matrix` for `MAT`? Otherwise I would suggest to test for `matrix` first.

Reply:
I think that most uses will be `Matrix` (a sparse matrix, actually), as these adjacency matrices are likely to be created with [PSMatch::makeAdjacencyMatrix()](https://rformassspectrometry.github.io/PSMatch/reference/adjacencyMatrix.html). For large data, these matrices are much smaller and faster to generate. Hence the appearance of `Matrix` here.